### ISSUE TITLE
feat: polls as a cast type

### ISSUE THREAD

**Issue by varunsrin** [farcasterxyz/hub#119]
**What is the feature you would like to implement?**
Allow polls where a user can post a question with multiple options and other users can make a choice. 

**Why is this feature important?**
A common user behavior that will be important in any public social network. It is particularly important to get right early because the data format is likely to be very different from a regular cast. 

**Will the protocol spec need to be updated??**
May need a new replication type in Section 4

**How should this feature be built? (optional)**
TBD

**Additional context**
TBD

**Comment by pfletcherhill** [comment]
@varunsrin are you sure this should be part of v2.0.0? Other than the profile metadata CRDT, I didn't think we wanted to add new message types for v2.0.0.

**Comment by varunsrin** [comment]
I don't think we should ship this in v2.0.0 but i think its worth designing now to understand how it might fit into a CRDT and then punting the implementation to post v2.0
